include/s3select.h

```diff
@@ -596,6 +621,16 @@ struct s3select : public bsc::grammar<s3select>
 
       trim_remove_type = (bsc::str_p("leading") | bsc::str_p("trailing") | bsc::str_p("both") );
 
+      datediff = (bsc::str_p("datediff") >> '(' >> (date_part)[BOOST_BIND_ACTION(push_date_part)] >> ',' >> arithmetic_expression >> ',' >> arithmetic_expression >> ')') [BOOST_BIND_ACTION(push_datediff)];
+
+      dateadd = (bsc::str_p("dateadd") >> '(' >> (date_part)[BOOST_BIND_ACTION(push_date_part)] >> ',' >> arithmetic_expression >> ',' >> arithmetic_expression >> ')') [BOOST_BIND_ACTION(push_dateadd)];
+
+      extract = (bsc::str_p("extract") >> '(' >> (date_part_extract)[BOOST_BIND_ACTION(push_date_part)] >> ',' >> arithmetic_expression >> ')') [BOOST_BIND_ACTION(push_extract)];
+
+      date_part = (bsc::str_p("year") | bsc::str_p("month") | bsc::str_p("day") | bsc::str_p("hour")  | bsc::str_p("minute") | bsc::str_p("second") );
```

Comment:
it better to add /BOOST_BIND_ACTION(push_date_part)/ in this rule(one time), and remove from other rules.